src/options.rs

```diff
@@ -409,15 +390,24 @@ impl Options {
             && atty::is(atty::Stream::Stdin);
 
         let mut builder = conn_params(&tmp.conn);
-        if let (Some(Command::Authenticate(auth)), None) = (&tmp.subcommand, &tmp.query) {
-            if builder.is_err() {
-                builder = Ok(Builder::new());
-                load_tls_options(&tmp.conn, builder.as_mut().unwrap())?;
+
+        // Special case for `edgedb instance link`
+        if let (Some(Command::Instance(i)), None) = (
+            &tmp.subcommand, &tmp.query
+        ) {
+            if let server::options::InstanceCommand::Link(
+                link
+            ) = &i.subcommand {
+                if builder.is_err() {
+                    builder = Ok(Builder::new());
+                    load_tls_options(&tmp.conn, builder.as_mut().unwrap())?;
+                }
+                server::link::prompt_conn_params(
```

Comment:
The only purpose of calling this function here is just to keep `tmp.conn` around, right? Let's just put `ConnectionOptions` into the final `Options` structure, and use it as a part of the normal command dispatch.

Reply:
Also reusing the password input/detection logic below.

Reply:
> Also reusing the password input/detection logic below.

It looks like that can be comfortably factored out to a function. I'm still feel uncomfortable when some commands are executed in the middle of the argument parsing (sans well-known ones like version and help).